### PR TITLE
Added Provisioning telemetry duration

### PR DESCRIPTION
**Pull request by brendandixon**
Correctly handle cleaning directories that include symlinks
Reduced polling interval #701
Enable selective agent testing
Reduced cloud-init files removed on deprovision #631

Signed-off-by: Brendan Dixon <brendandixon@me.com>

### CODE REVIEW

File: azurelinuxagent/common/utils/fileutil.py [hunk 1]
```diff
@@ -119,16 +119,20 @@ def rm_files(*args):
...
+            continue
+
+        for pp in os.listdir(p):
+            path = os.path.join(p, pp)
+            if os.path.isfile(path):
+                os.remove(path)
+            elif os.path.islink(path):
+                os.unlink(path)
+            elif os.path.isdir(path):
+                shutil.rmtree(path)
```
Comment (hglkrijger) [inline azurelinuxagent/common/utils/fileutil.py]: Won't this still fail if a sub-directory contains a symlink?

File: azurelinuxagent/ga/update.py [hunk 2]
```diff
@@ -918,3 +958,61 @@ def __str__(self):
...
+        d = ','.join(platform.linux_distribution())
+        for v in self.versions:
+            if re.match(v, d):
+                return True
+        return False
+
+    @property
+    def in_slice(self):
+        n = int((60 * self.slice) / 100)
+        return (n - datetime.utcnow().second) > 0
```
Comment (hglkrijger) [inline azurelinuxagent/ga/update.py]: should this be `>=` ?

File: azurelinuxagent/common/utils/fileutil.py (hunk 1 above)
Comment (brendandixon) [inline azurelinuxagent/common/utils/fileutil.py]: @hglkrijger Nope. It only fails if the directory passed to `shutil.rmtree` is a symlink.

File: azurelinuxagent/ga/update.py (hunk 2 above)
Comment (brendandixon) [inline azurelinuxagent/ga/update.py]: @hglkrijger Nope. Seconds range from 0-59 (that is, it is zero-based). Consider the 100% case: n will be 60...which will work for 0-59. :)

**Pull request merged**